src/styles/app-button/AppButton.scss

```diff
@@ -0,0 +1,139 @@
+$spinner-size: 1.5rem;
+$spinning-animation: spinning 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
+
+.app-button {
+  display: flex;
+  justify-content: center;
+  transition: all 0.3s ease;
+
+  &--primary {
+    background: #000;
+    border: 1px solid #000;
+    color: #fff;
+
+    &:hover {
+      opacity: 0.8;
+    }
+
+    &:focus,
+    &:active {
+      opacity: 0.7;
+    }
+
+    &.app-button--loading {
+      &:hover,
+      &:focus,
+      &:active {
+        opacity: 1;
+      }
+
+      .mini-spinner {
+        border-left: 3px solid #fff;
+        border-bottom: 3px solid #fff;
+        border-right: 3px solid #fff;
+        border-top: 3px solid rgba(255, 255, 255, 0.5);
+        border-radius: calc($spinner-size / 2);
+        width: $spinner-size;
+        height: $spinner-size;
+        box-sizing: border-box;
+        animation: $spinning-animation;
+      }
+    }
+
+    &.app-button--disabled {
+      opacity: 0.45;
+      cursor: not-allowed;
+
+      &:hover,
+      &:focus,
+      &:active {
+        opacity: 0.45;
+        cursor: not-allowed;
+      }
+    }
+  }
+
+  &--secondary {
+    background: transparent;
+    border: 1px solid #000;
+    color: #000;
+
+    &:hover {
+      background-color: rgba(153, 146, 134, 0.2);
+    }
+
+    &:focus {
+      background-color: rgba(140, 126, 98, 0.3);
+    }
+
+    &:active {
+      background-color: rgba(140, 126, 98, 0.3);
+      border-color: rgba(0, 0, 0, 0.75);
```

Comment:
Can we move all colors to own file and reuse them?

Reply:
Removed to file for variables